pkg/operator/ceph/object/zone/controller.go

```diff
@@ -219,43 +220,79 @@ func (r *ReconcileObjectZone) createCephZone(zone *cephv1.CephObjectZone, realmN
 		}
 	}
 
-	// check if master zone group does not exist yet for period
-	masterZone, err := decodeMasterZone(output)
+	// check if master zone does not exist yet for period
+	zoneGroupJson, err := object.DecodeZoneGroupConfig(output)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to parse `radosgw-admin zonegroup get` output")
+		return reconcile.Result{}, errors.Wrap(err, "failed to parse `radosgw-admin zonegroup get` output")
 	}
 
-	masterArg := ""
-	// master zone does not exist yet for zone group
-	if masterZone == "" {
-		masterArg = "--master"
+	// create zone
+	_, err = object.RunAdminCommandNoMultisite(objContext, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err == nil {
+		logger.Debugf("ceph zone %q already exists, new zone and pools will not be created", zone.Name)
+		return reconcile.Result{}, nil
 	}
 
-	// create zone
-	_, err = object.RunAdminCommandNoRealm(objContext, "zone", "get", realmArg, zoneGroupArg, zoneArg)
-	if err != nil {
-		if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
-			logger.Debugf("ceph zone %q not found, running `radosgw-admin zone create`", zone.Name)
-			_, err := object.RunAdminCommandNoRealm(objContext, "zone", "create", realmArg, zoneGroupArg, zoneArg, masterArg)
-			if err != nil {
-				return reconcile.Result{}, errors.Wrapf(err, "failed to create ceph zone %q", zone.Name)
-			}
-		} else {
-			return reconcile.Result{}, errors.Wrapf(err, "radosgw-admin zone get failed with code %d", code)
+	if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
+		logger.Debugf("ceph zone %q not found, running `radosgw-admin zone create`", zone.Name)
+
+		zoneIsMaster := false
+		if zoneGroupJson.MasterZoneID == "" {
+			zoneIsMaster = true
+		}
+
+		err = r.createPoolsAndZone(objContext, zone, realmName, zoneIsMaster)
```

Comment:
@travisn is it ok that this function is only returning an error and not a `reconcile.Result{}`. I could go ahead and not make this of type `ReconcilObjectZone` if that makes a difference.